odoo/addons/base/models/ir_model.py

```diff
@@ -1491,6 +1492,48 @@ def group_names_with_access(self, model_name, access_mode):
         """ % access_mode, [model_name])
         return [('%s/%s' % x) if x[0] else x[1] for x in self._cr.fetchall()]
 
+    @tools.ormcache('self.env.uid', 'self.env.su')
+    def _acl_map(self):
+        """ Return a mapping of model access rights as ``access[model_name][model]``. """
```

Comment:
What is the point of returning a mapping?  When I suggested this idea, what I had in mind was a refactoring of the method `check`.  But in this case, the only thing we need is the list of models the user can read.  Why not simply returning that list?

Reply:
>  But in this case, the only thing we need is the list of models the user can read. Why not simply returning that list?

1. because the intent is that if / once this is in master it can become the underlying implementation of `check` without having to touch it
2. because since this is cached we should have significantly more lookups than creations, and given we have hundreds of models this is way beyond lists being competitive with dicts